coramin/third_party/minlplib_tools.py

```diff
@@ -0,0 +1,336 @@
+import os
+from zipfile import ZipFile
+from pyomo.common import fileutils
+from pyomo.common import download
+import enum
+import math
+import csv
+from collections.abc import Iterable
+import logging
+
+
+logger = logging.getLogger(__name__)
+
+
+def get_minlplib_instancedata(download_dir=None):
+    """
+    Downlaod instancedata.csv from MINLPLib which can be used to get statistics on the problems from minlplib.
+
+    Parameters
+    ----------
+    download_dir: str
+        The directory in which to place the downloaded file. The default will be a 
+        directory called minlplib in the current working directory.
+    """
+    if download_dir is None:
+        download_dir = os.path.join(os.getcwd(), 'minlplib')
+
+    if not os.path.exists(download_dir):
+        os.makedirs(download_dir)
+
+    downloader = download.FileDownloader()
+    downloader.set_destination_filename(os.path.join(download_dir, 'instancedata.csv'))
+    downloader.get_binary_file('http://www.minlplib.org/instancedata.csv')
+
+
+def filter_minlplib_instances(instancedata_download_dir=None,
```

Comment:
Why take a dir and not just the file name?

Reply:
Good question. I changed this.